skimage/filters/thresholding.py

```diff
@@ -117,10 +122,22 @@ def try_all_threshold(image, figsize=(8, 5), verbose=True):
     """
     def thresh(func):
         """
-        A wrapper function to return a thresholded image.
+        Select a wrapper function to return a thresholded image. 
         """
-        def wrapper(im):
-            return im > func(im)
+        if ( func.__name__ == 'threshold_local' ):
+            # threshold_local requires the additional parameter 'block_size'
+            def wrapper(im):
+                return im > func(im, 15)
+        elif ( func.__name__ == 'threshold_phansalkar' ):
+            # threshold_phansalkar requires an equalized image
+            def wrapper(im):
+                im_eq = equalize_adapthist(im)
+                return im_eq > func(im_eq)
+        else:
+            # standard case
+            def wrapper(im):
+                return im > func(im)
+
         try:
             wrapper.__orifunc__ = func.__orifunc__
```

Comment:
Am I correct in assuming that this line will always raise an AttributeError? At least that's what local testing tells me. Assuming you are the author, @sciunto do you still remember the intention behind this line?

I may be a good idea to use [functools.wraps](https://docs.python.org/3/library/functools.html?highlight=wraps#functools.wraps) instead; `wraps` preserves the name of the function and makes the original one accessible through the `__wrapped__` attribute.

Reply:
Can you make a code suggestion? I just followed the existing pattern here. I wanted to avoid breaking anything.